### ISSUE TITLE
Bug in normalization procedure?

### ISSUE THREAD

**Issue by fzeiser** [oslocyclotronlab/ompy#89]
I first thought that there might be two problems, but I think the first one is so much of an issue afterall:

### 1. Prior limits
The current setup for the priors of the normalization looks like this:
https://github.com/oslocyclotronlab/ompy/blob/9ac0f5e24b2bd966dcff399cc1c245ae74d9cbc8/ompy/normalizer_simultan.py#L228-L253

Where the prior for `A` and `B` depends on the starting parameters obtained from the differential evolution minimization routine. I was slightly confused, because I wondered whether this has a negative impact when normalizing the whole ensemble. If the correct `B` for different ensemble members would be very different, do we have too much of a constraint choosing the same `B` for all of them with a Gaussian of 4 sigma?  
However, this is not the case. The starting parameters that influence the priors is chosen independently for each of the ensemble members! The only issue might be that one should probably relate the width of `B` to the uncertainty of `Gg`: If the uncertainty of `Gg` starts to be very large, the currently setup with a truncated normal with `sigma=4*mu` will limit the accepted `B`s to a smaller range

### 2. Error function
The error function currently is something like a chi^2 of the match to discretes and the high region (model that links to D0). I start to be a little unsure whether we should have `D0` as a parameter, or rather make `Eshift` the parameter of the minimization and a a term to the chi^2. This would be `(D0_exp - D0_model)^2 / sigma_D0`, where `D0_model` is obtained from the model for the higher energies. [Or to compare the nld at Sn directly]  

https://github.com/oslocyclotronlab/ompy/blob/9ac0f5e24b2bd966dcff399cc1c245ae74d9cbc8/ompy/normalizer_nld.py#L507-L533

**Comment by fzeiser** [comment]
I proceeded on understanding this one here. I'll still have to work on how to implement it though. My first though was to extend the errorfn to explicity take into account the uncertainty of D0:

``` py
def errfn(A, alpha, T, eshift, ...):
    err_low = f1(discretes; A, alpha; sigma_nld)...
    err_high = f2(A, alpha, T, eshift; ; sigma_nld)
    err_D0 = f3(D0, sigma_D0; A, alpha)
    return err_low +  err_high + err_D0

```

For now, the normalization will treat the discrete levels the same way as the nld at Sn (from D0). So the impact that D0 has will depend on the uncertainty of D0 and the uncertainty of the nld at low energies, where it is compared to the discrete levels. Now, there is two problems:
- we don't know the uncertainty of the NLD before it is normalized; so we can make up some mock values, like std= 10% of the value
- If a certain A and alpha gives a good match to the discrete levels, the effect that the D0 will have on A and alpha depends on the uncertainties of D0 and of the (mocked) uncertainty of the NLD. The match with D0 enters just as one more term in the chi^2. 

Shall I provide a small drawing of my thoughts?

I will head with Anders on how he would specif the likelihood. In principal we could maybe circumvent this problem by normalizing the impact of the chi2 of the discrete, the match to the higher region, and D0. But this needs to be done with some care.

**Comment by fzeiser** [comment]
Her the picture already, I’ll write the comment later

<img width="738" alt="PNG image" src="https://user-images.githubusercontent.com/15156776/70300627-a5dcd080-1800-11ea-917f-ae4434934a0f.png">

**Comment by fzeiser** [comment]
So, here the comment: I tried to see whether the normalization procedure works as I expected trying it on two extreme cases. The result surprised me, but having thought about it a little more, I maybe think that it could be fine the way it is.

So, the two cases were either stating a very small uncertainty for D0, or a very large uncertainty. A sketch of The results of the current minimization are depicted in Fig a, and b respectively. Whereas Fig a seemed reasonable to start with, I was surprised by b: If I state that D0 was quite unknown, I initially thought that the slope parameter should be very uncertain. This would be depicted in c. However, this was not the case and I think I understand why: If D0 has a large uncertainty, it is basically ignored in the chi^2 minimization, as it's weight is close to 0. The two transformation parameters A and alpha are then determined by the match to the discretes only. So if the shape of the oslo-method nld well describes the discrete states, the uncertainty on these parameters will be very small.  

Note: I did not sketch the high energy region which we use to determine the extrapolation parameters. But this is irrelevant for this question as they do not affect the transformation parameters.

Now it is difficult for me to judge whether this is a wanted/unwanted result. Maybe we can set this aside for now, but rather say that this is an idea for a later study: We can check how well the determination of the transformation parameters from the discrete states work. For this, we would take all datasets where we have oslo-method data and D0's. Then, we normalize the nld using only the discrete states. Next we find the postulated value of the nld at Sn from our data/extrapolation and compare it to the values we would get from using D0.

Finally I want to comment on another difficulty that remains. The effect of discrete states on determining the transformation parameters (and their uncertainties) now depends on two (three) aspects:
- The uncertainty that we assume the nld has at low energies where we compare it to the discrete states -> weighting in the chi^2. By default now, I set it to 10%, but this is arbitrary. 
- The number of points (bins with discrete states) taken into account in the fit. I'm not sure whether this is something we _want_. Somehow I guess that one could assume also, that the information of each type of dataset should be equally weighed, irrespective of the number of datapoints (so equal weights for i) discretes ii) D0). We'll have to think a little more about this I guess.
- (+ The uncertainty/weights of all other components of the chi2)
